### PR TITLE
Removed IPAddressField per deprecation timeline; refs #20439.

### PR DESCRIPTION
**Pull request by timgraham**

### CODE REVIEW

File: django/db/models/fields/__init__.py [hunk 1]
```diff
@@ -1892,40 +1892,15 @@ def formfield(self, **kwargs):
...
         'hint': 'Use GenericIPAddressField instead.',
-        'id': 'fields.W900',
+        'id': 'fields.E900',
     }
 
-    def __init__(self, *args, **kwargs):
-        kwargs['max_length'] = 15
-        super(IPAddressField, self).__init__(*args, **kwargs)
-
-    def deconstruct(self):
```
Comment (MarkusH) [inline django/db/models/fields/__init__.py]: this should probably stay, as we don't want `max_length` to suddenly show up somewhere in between states.

File: docs/ref/checks.txt [hunk 2]
```diff
@@ -97,6 +97,8 @@ Fields
 * **fields.W161**: Fixed default value provided.
 * **fields.W900**: ``IPAddressField`` has been deprecated. Support for it
```
Comment (MarkusH) [inline docs/ref/checks.txt]: I'll probably keep it, too. Just as historical documentation. You should put `E900` before `W900` though.

File: tests/migrations/test_deprecated_fields.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+class Tests(MigrationTestBase):
+    """
+    Deprecated model fields should still be usable in historic migrations.
+    """
+    @override_settings(MIGRATION_MODULES={"migrations": "migrations.deprecated_field_migrations"})
+    def test_migrate(self):
+        # Make sure no tables are created
+        self.assertTableNotExists("migrations_ipaddressfield")
+        # Run migration
+        call_command("migrate", verbosity=0)
```
Comment (MarkusH) [inline tests/migrations/test_deprecated_fields.py]: Now that `IPAddressField` doesn't have an `get_internal_type()` anymore, the created field is a varchar?

File: tests/model_fields/tests.py [hunk 4]
```diff
@@ -809,32 +809,6 @@ def test_IntegerField(self):
...
-        lazy_func = lazy(lambda: '127.0.0.1', six.text_type)
-        self.assertIsInstance(
-            IPAddressField().get_prep_value(lazy_func()),
-            six.text_type)
-        lazy_func = lazy(lambda: 0, int)
-        self.assertIsInstance(
-            IPAddressField().get_prep_value(lazy_func()),
-            six.text_type)
-
-    def test_IPAddressField_deprecated(self):
```
Comment (MarkusH) [inline tests/model_fields/tests.py]: Shouldn't we keep this test and test for `checks.Error`?

File: tests/migrations/test_deprecated_fields.py (hunk 3 above)
Comment (timgraham) [inline tests/migrations/test_deprecated_fields.py]: I think so, this is why I thought maybe we should keep it.

Comment (MarkusH) [inline tests/migrations/test_deprecated_fields.py]: Yea, good point.

**Pull request merged**